wasm/src/lazy_node.rs

```diff
@@ -35,6 +37,19 @@ impl LazyNode {
             _ => None,
         }
     }
+
+    #[wasm_bindgen]
+    pub fn to_bytes(&self, flag: u32) -> Option<Vec<u8>> {
+        let serializer = if (flag & ALLOW_BACKREFS) != 0 {
+            node_to_bytes_backrefs
+        } else {
+            node_to_bytes
```

Comment:
this case is risky. If you ever end up here on an untrusted `LazyNode`, you may blow up memory usage. The main reason we have `LazyNode` is to avoid serializing it to bytes

Reply:
That's something I hadn't thought of: compression mitigates this DOS vector. However, the compression code is quite complex and I don't know if there aren't DOS vectors there too. They're less obvious though. :)

Reply:
@arvidn How about adding `node_to_bytes_backrefs_limit` and update `node_to_bytes_backrefs` as:
```rust
pub fn node_to_bytes_backrefs(a: &Allocator, node: NodePtr) -> io::Result<Vec<u8>> {
    node_to_bytes_backrefs_limit(a, node, 2000000)
}
```

Reply:
As long as the case where the output is truncated is handled correctly, and not assumed to be an OK result. The function you suggest will silently ignore the failure though, which seems a bit risky.
Also, this comment was made on `node_to_bytes`, i.e. the version that doesn't support back-refs.